lib/profileAgents.js

```diff
@@ -0,0 +1,356 @@
+    controller: {
+      id: capabilityAgent.id,
+      seed: secret,
+      keystore: keystore.id
+    },
+    keystore: paKeystore.id,
+    capabilityInvocationKey: (await key.getKeyDescription()).id
+  };
+  const now = Date.now();
+  const meta = {created: now, updated: now};
+  let record = {
+    meta,
+    profileAgent
+  };
+  try {
+    const collection = utils.getCollection('profile-profileAgent');
+    const result = await collection.insert(record, database.writeOptions);
+    record = result.ops[0];
+  } catch(e) {
+    if(!database.isDuplicateError(e)) {
+      throw e;
+    }
+    throw new BedrockError(
+      'Duplicate profile agent.',
+      'DuplicateError', {
+        public: true,
+        httpStatusCode: 409
+      }, e);
+  }
+  return {
+    ...record,
+    profileAgent: {
+      ...record.profileAgent,
+      profile: undefined,
+      capabilityAgent: paZcapAgent,
+      keystoreAgent: paKeystoreAgent
+    }
+  };
+};
+
+/**
+ * Get a Profile Agent
+ *
+ * @param {Object} id the id of the ProfileAgent.
+ *
+ * @return {Promise<ProfileAgent>} resolves to a ProfileAgent.
+ */
+exports.get = async ({id}) => {
+  assert.string(id, 'id');
+
+  const query = {'profileAgent.id': id};
+  const collection = utils.getCollection('profile-profileAgent');
+  const record = await collection.findOne(query, {_id: 0});
+  if(!record) {
+    const details = {
+      profileAgent: id,
+      httpStatusCode: 404,
+      public: true
+    };
+    throw new BedrockError(
+      'Profile agent not found.',
+      'NotFoundError', details);
+  }
+  const {capabilityAgent, keystoreAgent} = await _getAgents(record);
+  return {
+    ...record,
+    profileAgent: {
+      ...record.profileAgent,
+      capabilityAgent,
+      keystoreAgent
+    }
+  };
+};
+
+/**
+ * Get all Profile Agent(s)
+ *
+ * @param {Object} account the account associated with the ProfileAgent(s).
+ *
+ * @return {Promise<Arrray<ProfileAgent>>} resolves to a ProfileAgent(s).
+ */
+exports.getAll = async ({account}) => {
+  assert.string(account, 'account');
+
+  const query = {'profileAgent.account': account};
+  const collection = utils.getCollection('profile-profileAgent');
+  const records = await collection.find(query, {_id: 0}).toArray();
+  if(records.length === 0) {
+    return [];
+  }
+  // TODO: Find proper promise-fun library for concurrency
+  const promises = records.map(async record => {
+    const {capabilityAgent, keystoreAgent} = await _getAgents(record);
+    return {
+      ...record,
+      profileAgent: {
+        ...record.profileAgent,
+        capabilityAgent,
+        keystoreAgent
+      }
+    };
+  });
+  return Promise.all(promises);
+};
+
+/**
+ * Get a Profile Agent
+ *
+ * @param {Object} profile the profile associated with the ProfileAgent.
+ * @param {Object} account the account associated with the ProfileAgent.
+ *
+ * @return {Promise<ProfileAgent>} resolves to a ProfileAgent.
+ */
+exports.getByProfile = async ({account, profile}) => {
```

Comment:
it would be good to add {..} = {} to all of these apis so that the assertions work as expected producing meaningful errors instead of destructuring errors.